core/client/client.go

```diff
@@ -80,11 +85,12 @@ var (
 // - - Sign key update requests.
 type Client struct {
 	*Verifier
-	cli        pb.KeyTransparencyClient
-	domainID   string
-	mutator    mutator.Func
-	RetryDelay time.Duration
-	trusted    types.LogRootV1
+	cli         pb.KeyTransparencyClient
+	domainID    string
+	mutator     mutator.Func
+	RetryDelay  time.Duration
+	trusted     types.LogRootV1
+	trustedLock sync.Mutex
```

Comment:
Good question.  

I'm using `trustedLock` as a transaction. The task is to ensure that each update to trusted is verified consistent with the previous value of trusted. The update process is 
1. Read `trusted`
2. Get consistency proof
3. Verify consistency proof
4. Write `trusted`

If two update processes run concurrently, we don't want A->C update overwritten by a A->D update because the proof fetched for A->D doesn't verify the consistency between C->D.  

`trusted` is sometimes used for other non-write purposes, where no lock is needed (I think).

Reply:
Could you elaborate why no lock is needed in non-write purposes? Do you assume no contention between readers and writers (for example, if `trusted` is never read while functions like `VerifiedGetLatestEpoch` are invoked), or you assume that the data will be updated atomically (which is not true I think)?

Reply:
Thanks for pushing back on this. I'll add the lock for every access for consistency.